eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/Argument.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.core.Box;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public class Argument<T> extends AbstractCachingOptional<T>
+{
+    // TODO Worth implementing here, not using the abstract class
```

Comment:
About implementing the `Optional`s directly instead of using the abstract class - I will rewrite all these classes later as it makes sense to optimize them since they will be core, often used classes.